design-proposals/klip-38-variable-substitution.md

```diff
@@ -264,7 +264,21 @@ Scope: SESSION
 In Java API client, the session is opened when `Client.create()` is called, and closed when `Client.close()` is called. Variables
 may be defined using a client method instead of a new syntax. This is to ensure users know the scope starts after the `create()` call.
 
-TBD: Define a Java API method for variable substitution.
+Users will define variables using the functions, `define(variable_name, value)` and `undefine(variable_name)`.
+The client will keep track of the calls made in a map. Alternatively, the `executeStatement` and `executeQuery`
```

Comment:
Don't we also want to support variable substitution in other methods as well, such as `insertInto()`, `streamInserts()`, and `createConnector()`, just to name a few? I don't think it makes sense to essentially double the number of methods on the Java client in order to add versions that accept a map of variables, so I'm a fan of `define()` and `undefine()` instead.